### PR TITLE
new API /metric_name/search:

### PR DESCRIPTION
**Pull request by ChandraAddala**
- Improving performance on the elastic search query used.

I had documented all my analysis on the wiki below.
https://one.rackspace.com/display/cloudmetrics/Blueflood+finder+performance+issues#Bluefloodfinderperformanceissues-Furthertuning

### CODE REVIEW

File: blueflood-elasticsearch/src/main/java/com/rackspacecloud/blueflood/io/AbstractElasticIO.java [hunk 1]
```diff
@@ -247,12 +247,16 @@ private int getTotalTokens(String query) {
...
      *                  "field" : "metric_name",
      *                  "include": "<regex>",
+     *                  "execution_hint": "map",
      *                  "size": 0
      *              }
      *          }
      *      }
      *  }
      *
+     * Execution hit of "map" works by using field values directly instead of ordinals
```
Comment (shintasmith) [inline blueflood-elasticsearch/src/main/java/com/rackspacecloud/blueflood/io/AbstractElasticIO.java]: typo: is it hit or hint?

Comment (ChandraAddala) [inline blueflood-elasticsearch/src/main/java/com/rackspacecloud/blueflood/io/AbstractElasticIO.java]: Its a typo. I fixed it now.

**Comment by usnavi** [comment]
+1

**Comment by shintasmith** [comment]
lgtm :+1:

**Pull request merged**